### ISSUE TITLE
Can't create a new character while dead/before respawning

### ISSUE THREAD

**Issue by Krzeszny** [Regalis11/Barotrauma#9084]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [X] My issue happened while using mods.

### What happened?

The character selection menu isn't shown in the TAB menu when dead.

I'm 99% sure it's not mod-related.
When you join a server and aren't spawned yet, you can't creat a new character.
Recently I've noticed you also can't do it when your character has died.

### Reproduction steps

1. Join an ongoing multiplayer game and spectate or have someone kill your character
2. Open the TAB menu
3. You can't create a new character before spawning

### Bug prevalence

Happens regularly

### Version

0.17.15.0

### Which operating system did you encounter this bug on?

Windows

### Relevant error messages and crash reports
![Barotrauma_nm37iBF0X8](https://user-images.githubusercontent.com/10529134/167273788-e67913f1-261c-4fa5-9843-f3a37ba314c0.png)

**Comment by Ek-Videogames** [comment]
This never doesn't happen - basically just the game is in a weird state when you die in the middle of the round
*(probably at least in-part a holdover from the old "respawn with penalty" system)*
so it hides the character panel from the tab menu - even though that is the GUI required to create a new character

This issue basically started when the talent tree was added with this updated GUI - before you could edit character mid round regardless of being alive or dead.

**Comment by Regalis11** [comment]
> This issue basically started when the talent tree was added with this updated GUI - before you could edit character mid round regardless of being alive or dead.

Oh, it's been in the game this long? I'm glad we were made aware of it now, I'll take a look!

**Comment by Regalis11** [comment]
Fixed in https://github.com/Regalis11/Barotrauma-development/commit/75100cb4d0479cf4556ce5d813ff9d8e505ebe2e

**Comment by Rokvach** [comment]
Seems like you aren't able to create a new character at all now

1. Launch server
2. Set respawn timer to 10 seconds
3. Launch mission mode
4. Press Tab and go to character tab and create a new character
5. Kill your current character
6. Notice that when you respawn you don't respawn as your new character
7. Die again
8. Still no changes
9. Return to lobby and notice that now your character info is using the new character on the lobby

**Comment by Regalis11** [comment]
Fixed in https://github.com/Regalis11/Barotrauma-development/commit/d3c8f9cffd8cc5765bf8382486c2ab2375d62a9a

**Comment by Rokvach** [comment]
I'm unable to rename my character still, I can change the class and look of the character but can't change the name. 

1. Start a multiplayer mission in mission mode
2. Open Tab and go to character page and create a new character
3. Rename your character and change the look and class
4. Kill your character
5. Open Tab character page and notice that your character does have the new name also notice that the Crew page still uses the old name
6.  Respawn your character
7. Notice that they have the old name but class and appearance is different

Additional issue on step 5, if you keep re-opening the tab menu on the character page you will notice that your characters class will change each time you re-open it while your character is dead. 

Another issue is that the character tab button to create a new character says "Settings" after the first time you've created a new character during the same round.

**Comment by Regalis11** [comment]
> Another issue is that the character tab button to create a new character says "Settings" after the first time you've created a new character during the same round.

That is actually intentional: you've already created a new character, and the button now edits that character. Might make sense to change that though if it's confusing.

**Comment by Rokvach** [comment]
>  you've already created a new character, and the button now edits that character

I had already spawned in as the new character so I would expect it to say create a new character as I would be then creating a new character of the newly spawned in character

**Comment by Regalis11** [comment]
Fixed all 3 issues in https://github.com/Regalis11/Barotrauma-development/commit/a12af2277b4c669706b493fbb0caa124508dcceb

> - Fixed setting a new name for the character not having an effect.
> - Fixed button in the character name saying "settings" indefinitely after you create a new character (should say "new character" after the character spawns).
> - Fixed character menu displaying a random job when you've opted to create a new character (+ had to touch some code related to the portrait clothing to address this, and I went ahead and removed that logic because it's no longer used).

**Comment by Rokvach** [comment]
The name is correct everywhere else except in the Tab crewlist its still using the old name, also if that player disconnects the disconnect/join message uses the old name still. The name doesn't seem to update in the crewlist and server messages even when you transition to a next level.

**Comment by Regalis11** [comment]
Fixed in https://github.com/Regalis11/Barotrauma-development/commit/b32ca799c6fd43b7f132819b42705937ce69afc0.

This might need some further thought though: it might be inconsistent with how the clients' names are tied to the characters in other situations.
- How should it work when you join a campaign where you have an existing character with a different name than what you're using now? Should it force the character's name to the name of the client, the other way around, or neither (which is how it works now)?
- When you opt to create a new character with a different name, at what point should we change the clients name? I made it change when the client spawns as that character, but not sure if that's the best way to go.
- Where should we display the client's name, and where the character's name, if they can be different?

Also something to note for testing: you can also create a new character even if your old character hasn't died yet, and in that case I think we definitely should wait before that character spawns before changing the client's name. + I also noticed a bug related to this, see https://github.com/Regalis11/Barotrauma/issues/9280

**Comment by Rokvach** [comment]
Yes the name should only be changed when the new character has spawned in, imo the renaming in the tab menu should functionally be the exact same as renaming your character in the lobby or in the server browser rather than just be a name for a specific character. 

I did notice that on mission mode if I created a new character with a new name and then killed my current character it still shows the old name on the tab menu crew list when I respawn as the new character. In campaign if I transition to a new level the name does get changed everywhere (haven't tested killing my current character in campaign mode yet)

I did notice that if you restart your server you are given your old name again, but not sure if this is what we want to happen

> How should it work when you join a campaign where you have an existing character with a different name than what you're using now? Should it force the character's name to the name of the client, the other way around, or neither (which is how it works now)?

This I'm not so sure about, I think it should use the name which you had while playing on that save but that feels like it contradicts the suggestion I made above where the name would be a global name rather than just a specific characters name, but if we force a name change to the currently used name for old saves it might ruin some roleplay if you have given a specific name for your character on that campaign, which you wouldn't use otherwise.
